app/src/main/java/de/westnordost/streetcomplete/quests/note_discussion/NoteDiscussionForm.java

```diff
@@ -162,11 +229,61 @@ private void onClickOk()
 
 		Bundle answer = new Bundle();
 		answer.putString(TEXT, noteText);
+		answer.putStringArrayList(IMAGE_PATHS, imagePaths);
 		applyImmediateAnswer(answer);
 	}
 
+	private void takePhoto()
+	{
+		Intent takePhotoIntent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+		if (takePhotoIntent.resolveActivity(getActivity().getPackageManager()) != null) {
+			try {
+				photoFile = createImageFile();
+			} catch (IOException e) {
+			}
+			if (photoFile != null) {
+				if (Build.VERSION.SDK_INT > 21) {
+					takePhotoIntent.putExtra(MediaStore.EXTRA_OUTPUT, getUriForFile(getActivity(), "de.westnordost.streetcomplete.fileprovider", photoFile));
+				} else {
+					takePhotoIntent.putExtra(MediaStore.EXTRA_OUTPUT, Uri.fromFile(photoFile));
+				}
+				startActivityForResult(takePhotoIntent, REQUEST_TAKE_PHOTO);
+			}
+		}
+	}
+
+	@Override
+	public void onActivityResult(int requestCode, int resultCode, Intent data) {
+		if (requestCode == REQUEST_TAKE_PHOTO) {
+			if (resultCode == RESULT_OK)
+			{
+				imageBitmaps.add(BitmapFactory.decodeFile(photoFile.getAbsolutePath()));
+				imagePaths.add(photoFile.toString());
+			} else
+			{
+				if (photoFile.exists())
+				{
+					photoFile.delete();
+				}
+			}
+		}
+	}
+
+	private File createImageFile() throws IOException
+	{
+		String timeStamp = new SimpleDateFormat("yyyyMMdd_HHmmss").format(new Date());
+		String imageFileName = "JPEG_" + timeStamp + "_";
+		File directory = getActivity().getExternalFilesDir(Environment.DIRECTORY_PICTURES);
+		return File.createTempFile(
+				imageFileName,
+				".jpg",
+				directory
+		);
+	}
+
```

Comment:
**Code duplication issue:** `LeaveNoteDialog` and `NoteDiscussionForm` share the same component - this "take photo" button plus the gridview that displays the photos already taken.

This should be put into a separate class.

Since you start an activity for result there, I propose that you put this functionality into a fragment and then add that fragment in the layout xml for both places.